lib/spack/spack/package.py

```diff
@@ -524,6 +524,12 @@ def __init__(self, spec):
         # a binary cache.
         self.installed_from_binary_cache = False
 
+        # Ensure that only one of these two attributes are present
+        if getattr(self, 'url', None) and getattr(self, 'urls', None):
```

Comment:
That's not exactly the same thing - `hasattr` looks if an attribute is there or not, `getattr` gets it and the test is done on the value being trueish. Maybe it's overly defensive, but I did it like that so that if you inherit from a package that e.g. defines `url` and you want to define `urls` you can in principle set `url` to `None` and define `urls`.